pkg/reconciler/pipelinerun/timeout.go

```diff
@@ -81,20 +81,29 @@ func init() {
 	}
 }
 
+// setPipelineRunTimeoutCondition sets the status of the PipelineRun to timed out.
+func setPipelineRunTimeoutCondition(ctx context.Context, pr *v1beta1.PipelineRun) {
+	pr.Status.SetCondition(&apis.Condition{
+		Type:    apis.ConditionSucceeded,
+		Status:  corev1.ConditionFalse,
+		Reason:  v1beta1.PipelineRunReasonTimedOut.String(),
+		Message: fmt.Sprintf("PipelineRun %q failed to finish within %q", pr.Name, pr.PipelineTimeout(ctx).String()),
+	})
+}
+
+// isPipelineRunTimeoutConditionSet returns true when the pipelinerun has the pipelinerun timed out reason
+func isPipelineRunTimeoutConditionSet(pr *v1beta1.PipelineRun) bool {
+	condition := pr.Status.GetCondition(apis.ConditionSucceeded)
+	return condition.IsFalse() && condition.Reason == v1beta1.PipelineRunReasonTimedOut.String()
+}
+
 // timeoutPipelineRun marks the PipelineRun as timed out and any resolved TaskRun(s) too.
 func timeoutPipelineRun(ctx context.Context, logger *zap.SugaredLogger, pr *v1beta1.PipelineRun, clientSet clientset.Interface) error {
 	errs := timeoutPipelineTasks(ctx, logger, pr, clientSet)
```

Comment:
Note: this would not use extra space in etcd for the pipelinerun

Reply:
👍